src/x86_64/interrupt.c

```diff
@@ -228,66 +227,83 @@ void common_handler()
     context f = ci->running_frame;
     int i = f[FRAME_VECTOR];
 
+    if (i >= n_interrupt_vectors) {
+        console("\nexception for invalid interrupt vector\n");
+        goto exit_fault;
+    }
+
+    // if we were idle, we are no longer
+    atomic_clear_bit(&idle_cpu_mask, ci->id);
+
+    int_debug("[%2d] # %d (%s), state %s, frame %p, rip 0x%lx, cr2 0x%lx\n",
+              ci->id, i, interrupt_names[i], state_strings[ci->state],
+              f, f[FRAME_RIP], f[FRAME_CR2]);
+
+    /* enqueue an interrupted user thread, unless the page fault handler should take care of it */
+    // what about bh?
+    if (ci->state == cpu_user && i >= INTERRUPT_VECTOR_START) {
```

Comment:
Not sure I understand here. But it would be nice if there was a frame return option for the fast path (nothing else to schedule)... In general we kind of need to clarify our policy with respect to preemption, as we're doing it kind of haphazardly and without any notion of quanta.